app/controllers/users_controller.rb

```diff
@@ -0,0 +1,35 @@
+class UsersController < ApplicationController
+  before_action :authenticate_user!
+
+  def create
+    User.create(user_params)
+  end
+
+  def edit
+    @user = current_user
+    @shipping_address = @user.shipping_address
+  end
+
+  def update
+    current_user.update(user_params)
+
+    shipping_address = ShippingAddress.find_or_initialize_by(user_id: current_user.id)
```

Comment:
check, shipping_address_params is not empty hash. if `shipping_address_params` is empty hash, don't need to check `ShippingAddress`